test/alphabet/union_alphabet.cpp

```diff
@@ -44,34 +44,155 @@
 
 using namespace seqan3;
 
-TEST(union_alphabet_test, default_ctr)
+TEST(union_alphabet_test, default_constructor)
 {
     using alphabet_t = union_alphabet<dna4, gap>;
-    alphabet_t{};
+    constexpr alphabet_t letter1{};
+
+    EXPECT_EQ(letter1.value, 0);
+}
+
+TEST(union_alphabet_test, base_alphabet_constructor)
+{
+    using alphabet_t = union_alphabet<dna4, dna5, gap>;
+
+    constexpr alphabet_t letter0{dna4::A};
+    constexpr alphabet_t letter1{dna4::C};
+    constexpr alphabet_t letter2{dna4::G};
+    constexpr alphabet_t letter3{dna4::T};
+    constexpr alphabet_t letter4{dna5::A};
+    constexpr alphabet_t letter5{dna5::C};
+    constexpr alphabet_t letter6{dna5::G};
+    constexpr alphabet_t letter7{dna5::T};
+    constexpr alphabet_t letter8{dna5::N};
+    constexpr alphabet_t letter9{gap::GAP};
+
+    EXPECT_EQ(letter0.value, 0);
```

Comment:
`value` is a public exposed member variable, so we have to test it and expect that users are using it. It makes no sense to be like, we don't document it so it is not there.